src/test/scala/org/constellation/storage/transactions/PendingTransactionsMemPoolTest.scala

```diff
@@ -151,6 +153,66 @@ class PendingTransactionsMemPoolTest extends FreeSpec with IdiomaticMockito with
 
       ret shouldBe None
     }
+
+    "it should return a consecutive chain of transactions and all the test transactions" in {
+      val memPool = PendingTransactionsMemPool[IO](txChainService, rl)
+
+      val tx1 = manuallyCreateTransaction("a", "", 0, false)
+      val tx2 = manuallyCreateTransaction("a", tx1.transaction.hash, tx1.transaction.ordinal, true)
+      val tx3 = manuallyCreateTransaction("a", tx2.transaction.hash, tx2.transaction.ordinal, false)
+      val tx4 = manuallyCreateTransaction("a", tx3.transaction.hash, tx3.transaction.ordinal, false)
+      val tx5 = manuallyCreateTransaction("a", tx4.transaction.hash, tx4.transaction.ordinal, true)
+
+      //not sending tx4 to pending to break the chain
+      (memPool.put(tx1.hash, tx1) >>
+        memPool.put(tx2.hash, tx2) >>
+        memPool.put(tx3.hash, tx3) >>
+        memPool.put(tx5.hash, tx5)).unsafeRunSync
+
+      val result = memPool.pull(10).unsafeRunSync
+
+      result shouldBe List(tx1, tx2, tx3, tx5).some
+    }
+  }
+
+  "takeConsecutiveTransactions" - {
+    val memPool = PendingTransactionsMemPool[IO](txChainService, rl)
+    val takeConsecutiveTransactions = PrivateMethod[List[TransactionCacheData]]('takeConsecutiveTransactions)
```

Comment:
why you cannot test consecutive transactions using public interface of pending tx mem pool? you don't need to test this private method, we should rely only on the API/public interface.
just put transactions to mem pool and make assertions what you get from `pull` method

Reply:
I guess that's just me being used to going from the ground up. I think it is easier and more edge cases can be found when you are testing smaller bits of code. Then when moving upwards only testing what has not been tested yet or even repeating test cases in the bigger picture (didn't do that here - true). That's just the approach I followed so far to be sure that changes I introduce work. But I get your point of testing the public API only and I think it's valid. If someone wanted to reimplement functionality of takeConsecutiveTransactions with other function, or nest it in public function where it's being used then that person would also need to move the test cases to check the public function instead. On the other hand I think that testing is not only for asserting on public API functionality but also for the developer to check if he implemented the code the way he intended - by testing that code out of context like this private method here that can be achieved. Anyway... ;) I will move the cases to be tested through public API, but I can't assure that we will not have this conversation again as for some reason going from the ground up is more natural and intuitive and I do it involuntarily ;).

Reply:
I work in the same way but try to think what you should expect at the end, not how to implement a single step. It’s like a declarative vs imperative - not how but what. And based on that you can create test cases upfront and work on the implementation then to fulfill those cases